lindi/LindiH5ZarrStore/LindiH5ZarrStoreOpts.py

```diff
@@ -0,0 +1,16 @@
+from typing import Union
+from dataclasses import dataclass
+
+
+@dataclass
```

Comment:
```suggestion
@dataclass(frozen=True)
```

Super minor, but I feel uneasy seeing `LindiH5ZarrStoreOpts()` as a default value in functions such as `LindiH5pyFile.from_hdf5_file` and `LindiH5ZarrStore.from_file` because it could lead to [strange bugs](https://docs.python-guide.org/writing/gotchas/), even though here, a user never has access to the stored opts to mutate it. It's just that the pattern makes me go 🤔 when I see it. I suggest either making this dataclass immutable or using `None` as the default value to protect against accidental mutation in the future.

Reply:
Thanks for the tip. I'll do both.